src/linalg.jl

```diff
@@ -5,6 +5,7 @@ for f in (:adjoint, :eigvecs, :inv, :pinv, :transpose)
 end
 
 LinearAlgebra.diag(B::BlockDiagonal) = map(i -> getindex(B, i, i), 1:minimum(size(B)))
+LinearAlgebra.diag(B::BlockDiagonal{T, V, true}) where {T, V} = mapreduce(diag, vcat, B.blocks)
```

Comment:
this is the first use case
```julia
julia> @btime diag(b);
  20.859 μs (199 allocations: 137.12 KiB)

julia> @btime diag(b);
  67.951 μs (298 allocations: 139.84 KiB)
```

Reply:
what are these timings showing us? is the second timing using the old version of the package?

Reply:
Yep, added the comment now.